include/inja/renderer.hpp

```diff
@@ -55,7 +55,20 @@ class Renderer : public NodeVisitor {
 
   void print_data(const std::shared_ptr<json> value) {
     if (value->is_string()) {
-      *output_stream << value->get_ref<const json::string_t&>();
+      std::string val;
+      if (config.escape_strings) {
+        // get the value as a dump() to properly escape values
+        val = value->dump();
+
+        // strip the leading and trailing " characters that are added by dump()
+        // if C++20 is adopted, val.starts_with and val.ends_with would clean this up a bit
+        val = val.substr(0,1) == "\"" && val.substr(val.length()-1,1) == "\""
+            ? val.substr(1, val.length()-2)
+            : val;
```

Comment:
one last thing i just noticed.. if this is already known to be a string, can we just get the string value instead of dumping as json?

i.e.
```
// store a string in a JSON value
json j_string = "this is a string";

// retrieve the string value
auto cpp_string = j_string.template get<std::string>();
```

Reply:
per readme in https://github.com/nlohmann/json

Reply:
The `.get<std::string>()` does not escape characters, whereas `.dump()` does.

Reply:
ah thanks, i misunderstood. lgtm